qemu/tests/virtio_fs_set_capability.py

```diff
@@ -0,0 +1,304 @@
+
+    fs_source = fs_params.get("fs_source_dir")
+    base_dir = fs_params.get("fs_source_base_dir",
+                             data_dir.get_data_dir())
+    if not os.path.isabs(fs_source):
+        fs_source = os.path.join(base_dir, fs_source)
+
+    host_data = os.path.join(fs_source, 'fs_test')
+
+    if os_type == "linux":
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, logging.info)
+        utils_misc.make_dirs(fs_dest, session)
+        error_context.context("Mount virtiofs target %s to %s inside"
+                              " guest." % (fs_target, fs_dest),
+                              logging.info)
+        if not utils_disk.mount(fs_target, fs_dest, 'virtiofs', session=session):
+            test.fail('Mount virtiofs target failed.')
+
+    else:
+        error_context.context("Start virtiofs service in guest.", logging.info)
+        exe_path = get_viofs_exe(session)
+        start_vfs_cmd = params["start_vfs_cmd"] % exe_path
+        session.sendline(start_vfs_cmd)
+
+        error_context.context("Check if virtiofs service is started.",
+                              logging.info)
+        check_virtiofs_cmd = params["check_virtiofs_cmd"]
+
+        if not utils_misc.wait_for(lambda: re.search("virtiofs",
+                                                     session.cmd_output(
+                                                         check_virtiofs_cmd),
+                                                     re.IGNORECASE), 30):
+            test.fail("Virtiofs service is failed to start.")
+
+        virtio_fs_disk_label = fs_target
+        error_context.context("Get Volume letter of virtio fs target, the disk"
+                              "lable is %s." % virtio_fs_disk_label,
+                              logging.info)
+        vol_con = "VolumeName='%s'" % virtio_fs_disk_label
+        vol_func = utils_misc.get_win_disk_vol(session, condition=vol_con)
+        volume_letter = utils_misc.wait_for(lambda: vol_func, 120)
+        fs_dest = "%s:" % volume_letter
+
+    guest_file = os.path.join(fs_dest, 'fs_test')
+    logging.info("The guest file in shared dir is %s.", guest_file)
+
+    try:
+        if cmd_set_trusted:
+            host_attributes = params["host_attributes"]
+            error_context.context("Set a trusted on guest.", logging.info)
+            session.cmd(cmd_yum_attr)
+            session.cmd(cmd_set_trusted)
+            output = session.cmd_output(cmd_get_trusted)
+            logging.info("The guest file trusted is %s.", output)
+            if 'trusted.test' not in output:
+                test.fail('Set trusted failed.')
+
+            process.run(cmd_yum_attr)
+            output = str(process.run('getfattr %s' % fs_source).stdout.strip())
+            logging.info("The host file trusted is %s.", output)
+            if host_attributes not in output:
+                test.fail('Set trusted failed.')
```

Comment:
Could you set the fail info with "It's failed to check the trusted info from host."

Reply:
Okay, update later.